File: beginner_contest_170/D_Not_Divisible/v1.cpp
#include<iostream>
#include<vector>
#include<algorithm>
using namespace std;

int main() {
    int n;
    cin >> n;
    vector<int> arr(n);
    for (int i = 0; i < n; i++) {
        cin >> arr[i];
    }

    sort(arr.begin(), arr.end());
    int arrMax = arr.back();
    vector<int> dp(arrMax + 1, true);

    int count = 0;
    for (int i = 0; i < n; i++) {
        int a = arr[i];
        if (dp[a]) {
            for (int j = 0; j * a <= arrMax; j++) {
                dp[j * a] = false;
            }
            if (i == n - 1 || a != arr[i + 1]) {
                count++;
            }
        }
    }
    cout << count;
}

File: beginner_contest_170/D_Not_Divisible/v1.js
const fs = require('fs')
const inputs = fs.readFileSync('/dev/stdin', 'utf8').trim().split(/\s+/).map(s => parseInt(s))

const n = inputs[0]
const arr = inputs.slice(1, n + 1).sort((a, b) => a - b)
const aMax = arr[arr.length - 1]

let count = 0
const dp = new Array(aMax + 1).fill(true)
arr.forEach((a, index) => {
  if (dp[a]) {
    for (let i = 1; i * a <= aMax; i++) {
      dp[i * a] = false
    }
    if (a !== arr[index + 1]) {
      count += 1
    }
  }
})
console.log(count)
